client/state/sites/reducer.js

```diff
@@ -105,6 +105,9 @@ export function items( state = {}, action ) {
 		case SITE_DELETE_RECEIVE:
 			return omit( state, action.siteId );
 
+		case SITE_REQUEST_FAILURE:
```

Comment:
This is excellent feedback! It felt wrong the way I was handling removal, but i couldn't think of an alternative. Thanks for the suggestion!